setup.cfg

```diff
@@ -53,6 +53,114 @@ test =
     hypothesis>=5.5.3
     pytest>=6.0
     pytest-xdist>=1.31
+    pytest-asyncio>=0.17.0
+# optional extras for recommended dependencies
+# see: doc/source/getting_started/install.rst
+performance =
+    bottleneck>=1.3.2
+    numba>=0.53.0
+    numexpr>=2.7.1
+timezone =
+    tzdata>=2022.1
+computation =
+    scipy>=1.7.1
+    xarray>=0.19.0
+aws =
+    boto3>=1.22.7
+    s3fs>=0.4.0
+gcp =
+    gcsfs>=2021.05.0
+    pandas-gbq>=0.15.0
+excel =
+    odfpy>=1.4.1
+    openpyxl>=3.0.7
+    pyxlsb>=1.0.8
+    xlrd>=2.0.1
+    xlwt>=1.3.0
+    xlsxwriter>=1.4.3
+parquet =
+    pyarrow>=1.0.1
+feather =
+    pyarrow>=1.0.1
+hdf5 =
+    blosc>=1.20.1
+    tables>=3.6.1
+spss =
+    pyreadstat>=1.1.2
+postgresql =
+    SQLAlchemy>=1.4.16
+    psycopg2>=2.8.6
+mysql =
+    SQLAlchemy>=1.4.16
+    pymysql>=1.0.2
+sql-other =
+    SQLAlchemy>=1.4.16
+html =
+    beautifulsoup4>=4.9.3
+    html5lib>=1.1
+    lxml>=4.6.3
+xml =
+    lxml>=4.6.3
+plot =
+    matplotlib>=3.3.2
+output_formatting =
+    jinja2>=3.0.0
+    tabulate>=0.8.9
+clipboard=
+    PyQt5>=5.15.1
+    qtpy>=2.2.0
+compression =
+    brotlipy>=0.7.0
+    python-snappy>=0.6.0
+    zstandard>=0.15.2
+# `all` supersets all the above options.
+# Also adds the following redundant, superseded packages that are listed as supported:
+#    fastparquet  (by pyarrow https://github.com/pandas-dev/pandas/issues/39164)
+# Other notes:
+#   fsspec is not explicitly listed below because it is a required dependency of s3fs, gcsfs (and we should let them
```

Comment:
So one argument for listing `fsspec` specifically, and may warrant having a `pandas[filesystem]` category independently, is if users are interacting with a remote filesystem that is not a major cloud-based one (gcsfs, s3fs, etc) e.g. `ftp://`, `file://`, etc

Reply:
I agree with @TomAugspurger; I also don't think fsspec should be a required dependency since its not all users will use IO functionality in pandas.

Reply:
Just a note that (last I looked) fsspec isn't used for HTTP(s) URLs.

Reply:
> I also don't think fsspec should be a required dependency since its not all users will use IO functionality in pandas.

We've just added in `pytz` as required in pandas `1.5.0` when not all users will interact with timezones and dates that require that package.  Where do we draw the line for "all users"?  I imagine we don't have exact guidelines...

**I'm happy with whatever packaging decision we can get quorum for.**

In that spirit I've added in a new extra for fsspec; it's an improvement over what we have now and I don't want to delay this PR to try and pitch fsspec as `required`.  

> Just a note that (last I looked) fsspec isn't used for HTTP(s) URLs.

Agreed, anything URL-like goes to fsspec, excluding `http(s)://`.  FWIW I didn't unpack further to understand the http route in pandas (although you certainly could use fsspec alongside requests).